SalesforceFoundation/k12-architecture-kit
# [K12] Test and Test Score Objects K12 Additions

# Critical Changes

# Changes
In version 1.8, we made the following changes. If you installed an earlier version or installed on top of an existing org, consider making them to your org.

- We introduced a K-12 All Tests list view on Test. If you choose to create a K-12 All Tests list view, we recommend this configuration:
    - Sharing Settings: All users can see this list view-
    - Filter by Owner: All tests
    - Display order: Test Name, Contact, Test Type, Test Date
      See [Create or Clone a List View](https://help.salesforce.com/articleView?id=customviews_lex.htm&type=5).
   - Create a K12 Test Compact Layout with fields in this order: Test Name, Contact, Test Type, Test Date

- We introduced a K-12 All Test Scores list view on Test Scores. If you choose to create a K-12 All Tests Scores list view, we recommend this configuration:
  - Sharing Settings: All users can see this list view
  - Filter by Owner: All test scores
  - Display order: Test Score Name, Test Type, Subject Area, Score, Percentile
    See [Create or Clone a List View](https://help.salesforce.com/articleView?id=customviews_lex.htm&type=5).
   - Add K-12 All Test Scores as a related list on Test.
     For more information, see [Display a New Custom Object After a Product Release](https://powerofus.force.com/s/article/EDA-Display-Cust-Obj-Post-Rel).
    - Create a K12 Test Compact Layout with fields in this order: Test Name, Contact, Test Type, Test Date

- We introduced new quick actions for Edit, New Note, Prinatble View, Change Owner, Clone, and Delete. If you want the New Case quick action to appear, you'll need to update the global quick action for New Case and change the record type as described in [Quick Action Considerations](https://help.salesforce.com/articleView?id=actions_considerations.htm&type=5).

# Issues Closed
Closes #177 
# New Metadata
Page Layouts:
- K12 Kit Test Layout
- K12 Kit Test Score Layout

List View:
- K12 All Tests
- K12 All Test Score

# Deleted Metadata

# Testing Notes
1. Trial Org: this should be tested only after the PR has been merged to master

2. Dev_Namespaced Shape + dev_org_namespaced flow: 
Pre-requisite: Go to Setup || User Interface || Global Actions || New Case
- Click Edit on "New Case"
- Update "RecordType" value from "Master" to "Incident" 
- Click save 

Steps for testing: 
- Create Test and Test Score records
- Test and Test Score page layout should match the screenshots on this quip doc https://salesforce.quip.com/5YeEArOU11bn#XSOACAgOHRH and https://salesforce.quip.com/5YeEArOU11bn#XSOACAVFf4P
- Test and Test Score list view should match the screenshots on this quip doc 
https://salesforce.quip.com/5YeEArOU11bn#XSOACAjlpPk and 
https://salesforce.quip.com/5YeEArOU11bn#XSOACAg0xPp

3. Dev_Namespaced Shape + qa_org flow: 
Pre-requisite: Go to Setup || User Interface || Global Actions || New Case
- Click Edit on "New Case"
- Update "RecordType" value from "Master" to "Incident" 
- Click save 

Steps for testing: 
- Create Test and Test Score records
- Test and Test Score page layout should match the screenshots on this quip doc https://salesforce.quip.com/5YeEArOU11bn#XSOACAgOHRH and https://salesforce.quip.com/5YeEArOU11bn#XSOACAVFf4P
- Test and Test Score list view should match the screenshots on this quip doc 
https://salesforce.quip.com/5YeEArOU11bn#XSOACAjlpPk and 
https://salesforce.quip.com/5YeEArOU11bn#XSOACAg0xPp
